src/basic/path-util.c

```diff
@@ -1441,3 +1441,35 @@ int path_glob_can_match(const char *pattern, const char *prefix, char **ret) {
                 *ret = NULL;
         return false;
 }
+
+const char* default_path(void) {
+#if HAVE_SPLIT_BIN
+        bool bad = false;
```

Comment:
`bad` sounds spurious to me. Also, why not to cache the result, like other system setup checkers?

Reply:
Yeah, the result should be cached. While the transition could be done without rebooting, I think having `sbin` in that case doesn't hurt.

Reply:
"bad" because it's … bad. In the sense that we want to deprecate split-bin in the future.

If we reload the manager after doing a live upgrade, we want to use the new path. Caching would make that impossible.

Reply:
(BTW, off-topic here, but speaking of deprecation has anyone looked into actually dropping sysv script support in the code? Do we want that for v257?)

Reply:
Should we support such case?? Why not reexec in such case? Even if it's necessary to support, I think then we should reset the cache in such case. I hope that should be necessary at most by PID1.